back-end/src/main/java/admin/auth/AuthenticationPrincipalConfig.java

```diff
@@ -2,27 +2,44 @@
 
 import admin.auth.application.AuthService;
 import admin.auth.ui.AuthenticationPrincipalArgumentResolver;
+import admin.auth.ui.LoginInterceptor;
 import java.util.List;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
 
 @Configuration
 public class AuthenticationPrincipalConfig implements WebMvcConfigurer {
 
     private final AuthService authService;
+    private final LoginInterceptor loginInterceptor;
 
-    public AuthenticationPrincipalConfig(AuthService authService) {
+    public AuthenticationPrincipalConfig(AuthService authService, LoginInterceptor loginInterceptor) {
         this.authService = authService;
+        this.loginInterceptor = loginInterceptor;
     }
 
     @Override
-    public void addArgumentResolvers(List argumentResolvers) {
+    public void addArgumentResolvers(List<HandlerMethodArgumentResolver> argumentResolvers) {
         argumentResolvers.add(createAuthenticationPrincipalArgumentResolver());
     }
 
     @Bean
     public AuthenticationPrincipalArgumentResolver createAuthenticationPrincipalArgumentResolver() {
         return new AuthenticationPrincipalArgumentResolver(authService);
     }
+
+    /*
+    //TODO
+    //- interceptor 적용시 주석해제
+    //- interceptor 적용할 path 설정
+    @Override
+    public void addInterceptors(InterceptorRegistry registry) {
+        registry.addInterceptor(loginInterceptor)
+                .excludePathPatterns("/api/login", "/api/labs/","/api/members/")
+                .addPathPatterns("/api/**");
+    }
```

Comment:
pattern 매칭 전략에 대해서 설명이 가능할까요?? 패턴 매칭에 대해 지식이 부족해서 몇가지 질문 남겨요! 제가 생각한 방식으로는 다음처럼 구현이 될 수 있을 것 같아요!
```suggestion
    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(loginInterceptor)
                .excludePathPatterns("/api/login", "/api/members/me")
                .addPathPatterns("/api/**");
    }
```
`/api/labs/` 부분을 exclude 한 이유가 궁금합니다!
`/api/members/` 에서 /api/members/me` 만을 한정하지 않은 이유도 궁금합니다.!

Reply:
기존에 auth를 주지 않고 테스트한 경우를 통과시키기 위함입니다 :) 해당 두 부분을 제외하면 기존 테스트 코드가 터져요..ㅎㅎ 일단 돌아가는 테스트 구현을 하기 위해 넣어주었고, 추후 인증 인가 정책이 확정되면 PathPattern에 내용들을 넣고 테스트 하면 좋을 것 같아요!

Reply:
기존 테스트코드에서 `/api/labs/` `/api/members/` POST를 사용하고 있습니다ㅎㅎㅎ

Reply:
혹시 배럴이 생각하는 최종 pattern 은 어떤 모습일까요??

Reply:
전 이부분이 잘 안 그려져서, 오늘 오전 회의때도 token 적용까지 지금 가능한지 모르겠다고 했던 것이었어요ㅎㅎ token을 적용하기 위해서는  
Manager 계정을 어떻게 할 것인지, Admin 권한은 어떻게 설정할 것인지 등을 알아야 한다고 생각했거든요! :) 

그래서 임의로 해당 기능을 구현만 하고 주석처리를 한 것이고, 적용은 또 다른 문제라고 생각했었습니다ㅎㅎ

DM 으로 요청주신 worker 에 대한 것은 exclude로 추가했어요! :) 혹시 더 필요한 내용이 있다고 생각되면 알려주세요~

Reply:
혹시 현재 그려지는 최종 모습이 있다면 공유 부탁드릴게요 🙏  그 베이스로 같이 고민해 보겠습니당!ㅎㅎㅎㅎㅎ

Reply:
아니면 API 정리가 다 되고 나면 이후에 이 부분을 적용해보는 것도 좋을 것 같아요 :) 결국 모든 적용할 구간에 테스트코드에서 Authorization 헤더를 넣어주어야 하니까요! 

생각보다 꽤 큰 작업일 것 같아서요.